pkgs/by-name/al/alda/package.nix

```
@@ -1,46 +1,84 @@
 {
   lib,
   stdenv,
-  fetchurl,
+  fetchFromGitHub,
+  buildGoModule,
+  gradle,
   makeWrapper,
   jre,
+  symlinkJoin,
 }:
-
-stdenv.mkDerivation (finalAttrs: {
+let
   pname = "alda";
   version = "2.3.1";
-
-  src_alda = fetchurl {
-    url = "https://alda-releases.nyc3.digitaloceanspaces.com/${finalAttrs.version}/client/linux-amd64/alda";
-    hash = "sha256-m4d3cLgqWmGMw0SM4J+7nvV/ytSoB7obMDiJCh3yboQ=";
+  src = fetchFromGitHub {
+    owner = "alda-lang";
+    repo = "alda";
+    rev = "release-${version}";
+    hash = "sha256-//VfegK8wkGKSpvtsNTEQqbVJkcucNiamoNIXaEBLb8=";
   };
 
-  src_player = fetchurl {
-    url = "https://alda-releases.nyc3.digitaloceanspaces.com/${finalAttrs.version}/player/non-windows/alda-player";
-    hash = "sha256-XwgOidQjnMClXPIS1JPzsVJ6c7vXwBHBAfUPX3WL8uU=";
-  };
+  alda_client = buildGoModule {
+    pname = "${pname}-client";
+    inherit version src;
 
-  dontUnpack = true;
+    sourceRoot = "${src.name}/client";
+    vendorHash = "sha256-h09w6ZLirLNxYv/ibeN5pCnXSvT+1FGiXiYNReZBMXI=";
 
-  nativeBuildInputs = [ makeWrapper ];
+    preBuild = ''
+      go generate main.go
+    '';
+
+    env.CGO_ENABLED = 0;
+    ldflags = [
+      "-w"
+      "-extldflags '-static'"
+    ];
+    tags = [ "netgo" ];
+    subPackages = [ "." ];
 
-  installPhase =
-    let
-      binPath = lib.makeBinPath [ jre ];
-    in
-    ''
-      install -D ${finalAttrs.src_alda} $out/bin/alda
-      install -D ${finalAttrs.src_player} $out/bin/alda-player
 
-      wrapProgram $out/bin/alda --prefix PATH : $out/bin:${binPath}
-      wrapProgram $out/bin/alda-player --prefix PATH : $out/bin:${binPath}
+    postInstall = ''
+      mv $out/bin/client $out/bin/alda
     '';
+  };
+  alda_player = stdenv.mkDerivation {
+    pname = "${pname}-player";
+    inherit version src;
+
+    sourceRoot = "${src.name}/player";
+    nativeBuildInputs = [ gradle makeWrapper ];
+
+    mitmCache = gradle.fetchDeps {
+      inherit pname;
+      data = ./deps.json;
+    };
+    __darwinAllowLocalNetworking = true;
+
+    gradleBuildTask = "fatJar";
+
+    installPhase = ''
+    mkdir -p $out/{bin,share}
+    cp build/libs/alda-player-fat.jar $out/share
+
+    makeWrapper ${lib.getExe jre} $out/bin/alda-player \
+      --add-flags "-jar $out/share/alda-player-fat.jar"
+  '';
+  };
+in
+symlinkJoin {
+  inherit pname version;
+  paths = [ alda_client alda_player ];
 
   meta = {
     description = "Music programming language for musicians";
     homepage = "https://alda.io";
     license = lib.licenses.epl10;
```

https://github.com/alda-lang/alda/commit/a6f213efdef127907c6d760c472c038abe753cb7
```suggestion
    license = lib.licenses.epl20;
```